Simplified/NYPLAccountSignInViewController.m

```diff
@@ -702,26 +704,80 @@ - (void)validateCredentials
   NSMutableURLRequest *const request =
     [NSMutableURLRequest requestWithURL:[NYPLConfiguration loanURL]];
   
-  // Necessary to support longer login times when using usernames.
   request.timeoutInterval = 20.0;
   
-  request.HTTPMethod = @"HEAD";
-  
   self.isCurrentlySigningIn = YES;
   NSURLSessionDataTask *const task =
     [self.session
      dataTaskWithRequest:request
-     completionHandler:^(__attribute__((unused)) NSData *data,
+     completionHandler:^(NSData *data,
                          NSURLResponse *const response,
                          NSError *const error) {
        
        self.isCurrentlySigningIn = NO;
-       // This cast is always valid according to Apple's documentation for NSHTTPURLResponse.
+
        NSInteger const statusCode = ((NSHTTPURLResponse *) response).statusCode;
        
-       // Success.
        if(statusCode == 200) {
+         
+#if defined(FEATURE_DRM_CONNECTOR)        
+
+         NYPLXML *loansXML = [NYPLXML XMLWithData:data];
+         NYPLOPDSFeed *loansFeed = [[NYPLOPDSFeed alloc] initWithXML:loansXML];
+         
+         if (loansFeed.licensor) {
+           [[NYPLAccount sharedAccount] setLicensor:loansFeed.licensor];
+         } else {
+           NYPLLOG(@"Login Failed: No Licensor Token received or parsed from OPDS Loans feed");
+           [self authorizationAttemptDidFinish:NO error:nil];
+           return;
+         }
+         
+         NSMutableArray *licensorItems = [[loansFeed.licensor[@"clientToken"] stringByReplacingOccurrencesOfString:@"\n" withString:@""] componentsSeparatedByString:@"|"].mutableCopy;
+         NSString *tokenPassword = [licensorItems lastObject];
+         [licensorItems removeLastObject];
+         NSString *tokenUsername = [licensorItems componentsJoinedByString:@"|"];
+         
+         NYPLLOG(@"***DRM Auth/Activation Attempt***");
+         NYPLLOG_F(@"\nLicensor: %@\n",loansFeed.licensor);
+         NYPLLOG_F(@"Token Username: %@\n",tokenUsername);
+         NYPLLOG_F(@"Token Password: %@\n",tokenPassword);
+         
+         [[NYPLADEPT sharedInstance]
+          authorizeWithVendorID:[[NYPLAccount sharedAccount] licensor][@"vendor"]
```

Comment:
Just continuing on the last point, this may cause a crash if the `vendor` field is not present when authorizing. It would be nice to go back to this at some point in the future and clean up the representation of licensors.